Mergin/plugin.py

```diff
@@ -86,40 +87,56 @@ def _login_error_message(self, e):
+
         target_dir = os.path.abspath(os.path.join(parent_dir, self.project['name']))
-        settings = QSettings()
 
-        QApplication.setOverrideCursor(Qt.WaitCursor)
-        try:
-            self.mc.download_project(self.project_name, target_dir)
-            settings.setValue('Mergin/localProjects/{}/path'.format(self.project_name), target_dir)
-            self.path = target_dir
-            self.setIcon(QIcon(os.path.join(icon_path, "folder-solid.svg")))
-            QApplication.restoreOverrideCursor()
-
-            msg = "Your project {} has been successfully downloaded. " \
-                  "Do you want to open project file?".format(self.project_name)
-            btn_reply = QMessageBox.question(None, 'Project download', msg,
-                                             QMessageBox.Yes | QMessageBox.No, QMessageBox.Yes)
-            if btn_reply == QMessageBox.Yes:
-                self.open_project()
-        except (URLError, ValueError) as e:
-            QgsApplication.messageLog().logMessage(f"Mergin plugin: {str(e)}")
-            QApplication.restoreOverrideCursor()
-            msg = "Failed to download your project {}.\n" \
-                  "Please make sure your Mergin settings are correct".format(self.project_name)
-            QMessageBox.critical(None, 'Project download', msg, QMessageBox.Close)
-        except LoginError as e:
-            self._login_error_message(e)
-        except Exception as e:
-            QApplication.restoreOverrideCursor()
-            msg = "Failed to download your project {}.\n" \
-                  "{}".format(self.project_name, str(e))
-            QMessageBox.critical(None, 'Project download', msg, QMessageBox.Close)
+        if os.path.exists(target_dir):
+            QMessageBox.warning(None, "Download Project", "The target directory already exists:\n"+target_dir+
+                                      "\n\nPlease select a different directory.")
+            return
+
+        dlg = SyncDialog()
+        dlg.download_start(self.mc, target_dir, self.project_name)
+
+        dlg.exec_()  # blocks until completion / failure / cancellation
+
+        if dlg.exception:
+            if isinstance(dlg.exception, (URLError, ValueError)):
+                QgsApplication.messageLog().logMessage("Mergin plugin: " + str(dlg.exception))
+                msg = "Failed to download your project {}.\n" \
+                      "Please make sure your Mergin settings are correct".format(self.project_name)
+                QMessageBox.critical(None, 'Project download', msg, QMessageBox.Close)
+            elif isinstance(dlg.exception, LoginError):
+                self._login_error_message(dlg.exception)
+            else:
+                msg = "Failed to download your project {}.\n" \
+                      "{}".format(self.project_name, str(dlg.exception))
+                QMessageBox.critical(None, 'Project download', msg, QMessageBox.Close)
```

Comment:
why we sometimes log to messageLog and sometimes not?

Reply:
No idea - I have merely copied over that error handling handling code. I think we don't need to write anything to the QGIS message log. I would keep it as is for now and remove later when we have also better logging in the python client...